src/nl/hannahsten/texifyidea/action/debug/PerformanceDiagnosticAction.kt

```diff
@@ -0,0 +1,79 @@
+package nl.hannahsten.texifyidea.action.debug
+
+import com.intellij.openapi.actionSystem.ActionUpdateThread
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.ui.DialogBuilder
+import kotlinx.html.body
+import kotlinx.html.h2
+import kotlinx.html.h3
+import kotlinx.html.html
+import kotlinx.html.stream.createHTML
+import kotlinx.html.table
+import kotlinx.html.td
+import kotlinx.html.tr
+import nl.hannahsten.texifyidea.index.LatexProjectStructure
+import javax.swing.JLabel
+import javax.swing.SwingConstants
+
+class PerformanceDiagnosticAction : AnAction() {
```

Comment:
> Also, diagnostic information about building the fileset is record. On my laptop, it shows that an average of < 20ms is only needed to build the fileset, except for the initial build which would take ~200ms. The overall performance seems to be good. 

Performance is amazing! I just tested on a real life project:

<img width="313" height="297" alt="{5130D079-8C38-4628-A5B1-0574625ECD0E}" src="https://github.com/user-attachments/assets/ff749277-d687-403f-8da0-4ca45caaab2c" />

After the initialization, performance of the fileset refresh is on average about 30-40 times faster than before on this project.